associated-token-account/program/src/instruction.rs

```diff
@@ -0,0 +1,51 @@
+//! Program instructions
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    pubkey::Pubkey,
+    sysvar,
+};
+
+use crate::{get_associated_token_address, id};
+
+/// Instructions supported by the AssociatedTokenAccount program
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum AssociatedTokenAccountInstruction {
+    /// Creates an associated token account for the given wallet address and token mint
+    ///
+    ///   0. `[writeable,signer]` Funding account (must be a system account)
+    ///   1. `[writeable]` Associated token account address to be created
+    ///   2. `[]` Wallet address for the new associated token account
+    ///   3. `[]` The token mint for the new associated token account
+    ///   4. `[]` System program
+    ///   5. `[]` SPL Token program
+    ///   6. `[]` Rent sysvar
```

Comment:
Please drop the Rent sysvar from the explicit Create instruction, as we can use `Rent::get()` from within the program instead now

Reply:
That would require a change from ATA to use `InitializeAccount3`, which hasn't been deployed yet :eyes:

Reply:
Yeah, I think it is the best to make the explicit `Create` instruction exactly the same as the implicit one to avoid confusion and hidden API and then we can add `Create2`

Reply:
Good point Jon.  Fortunately ATA has the same deployment method as Token, so when we redeploy ATA we'd do Token as well, which means that we don't need `Create2`

Reply:
I tried to use `InitializeAccount3`but it gives error `Error: Invalid instruction`. I'm guessing it happens because `ProgramTest` doesn't load the new token program yet. Is that right or I'm missing something?

Reply:
Oh right, ProgramTest doesn't include a version of SPL Token with `InitializeAccount3` because it's not deployed yet.
In short term we could use `ProgramTest::add_program()` to manually add the latest SPL Token program into the test environment though, then undo that change later after the next SPL Token upgrade.  

Commenting out the test for the  explicit Create instruction in the short term works too.

Reply:
I've got it working by conditionally loading the latest `spl_token` program for tests when `InitializeAccount3` is required.